import_service/api_gateway.js

```diff
@@ -17,13 +18,106 @@ class ApiGateway extends Stack {
     const {
       import_service_fn = aws_lambda,
     } = props;
-
+    
     const api = new aws_apigateway.RestApi(this, 'ImportApi', {
-        restApiName: 'Import Service'
+        restApiName: 'Import Service',
+        cloudWatchRole: true,
+        description: "This service import products for RSS AWS course.",
+        defaultCorsPreflightOptions: {
+          allowOrigins: aws_apigateway.Cors.ALL_ORIGINS,
+          allowMethods: aws_apigateway.Cors.ALL_METHODS,
+          allowHeaders: aws_apigateway.Cors.DEFAULT_HEADERS,
+          allowCredentials: true,
+        },
+    });
+
+    const basicAuthorizerArn = Fn.importValue("BasicAuthorizerArn");
+    const basicAuthorizerArnRole = Fn.importValue("BasicAuthorizerArnRole");
+    const basicAuthorizerRole = Role.fromRoleArn(
+      this,
+      "BasicAuthorizerRole",
+      basicAuthorizerArnRole
+    );
+    const basicAuthorizerLambda = aws_lambda.Function.fromFunctionAttributes(
+      this,
+      "authFunction",
+      {
+        functionArn: basicAuthorizerArn,
+        role: basicAuthorizerRole,
+      }
+    );
+
+    const authorizer = new aws_apigateway.TokenAuthorizer(this, 'APIGatewayAuthorizer', {
```

Comment:
Try to change [TokenAuthorizer ](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_apigateway/TokenAuthorizer.html)to [RequestAuthorizer](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_apigateway/RequestAuthorizer.html).